### PR TITLE
 add a transform challenge to the stack overflow track

### PR DESCRIPTION
**Pull request by hendrikmuhs**
add a transform challenge to the so track

to be run with: `--challenge=transform`

/CC @ywelsch

### CODE REVIEW

File: so/challenges/default.json [hunk 1]
```diff
@@ -50,4 +50,111 @@
...
+          "operation": {
+            "operation-type": "create-index",
+            "settings": {{index_settings | default({}) | tojson}}
+          }
+        },
+        {
+          "name": "check-cluster-health",
+          "operation": {
+            "operation-type": "cluster-health",
+            "index": "logs-*",
```
Comment (droberts195) [inline so/challenges/default.json:72]: Is it a mistake that we check the health of `logs-*` when the transform operates on the `so` index?

Comment (inqueue) [reply so/challenges/default.json:72]: https://github.com/elastic/elasticsearch/issues/89980 was filed to address the Cluster Health API behavior when the target does not exist.

**Review by inqueue — CHANGES_REQUESTED** [review]
The transform challenge runs into a heap wall when executed under defaults:

```shell
$ esrally race --track-path /Users/jbryan/dev/src/reviews/rally-tracks/so --challenge=transform

[INFO] Race id is [96360175-e41d-4d51-be59-677e239ab8fa]
[INFO] Preparing for race ...
[INFO] Downloading track data (8.9 GB total size)                                 [100.0%]
[INFO] Decompressing track data from [/Users/jbryan/.rally/benchmarks/data/so/posts.json.bz2] to [/Users/jbryan/.rally/benchmarks/data/so/posts.json] (resulting size: [33.12] GB) ... [OK]
[INFO] Preparing file offset table for [/Users/jbryan/.rally/benchmarks/data/so/posts.json] ... [OK]
[INFO] Racing on track [so], challenge [transform] and car ['defaults'] with version [8.5.0-SNAPSHOT].

Running delete-index                                                           [100% done]
Running create-index                                                           [100% done]
Running check-cluster-health                                                   [100% done]
Running index-append                                                           [100% done]
Running force-merge                                                            [100% done]
Running wait-until-merges-finish                                               [100% done]
Running delete-transform-group-by-user                                         [100% done]
Running create-transform-group-by-user                                         [100% done]
Running start-transform-group-by-user                                          [100% done]
Running wait-for-transform-group-by-user                                       [  1% done]
[ERROR] Cannot race. Error in load generator [0]
	Cannot run task [wait-for-transform-group-by-user]: Transform [transform-group-by-user] failed with [Insufficient memory for search after repeated page size reductions to [10], unable to continue pivot, please simplify job or increase heap size on data nodes.].
```

1g heap is the default. I'm trying again with car `4gheap` to see if makes a difference. I think we will want to at least add [test mode support](https://esrally.readthedocs.io/en/latest/adding_tracks.html?highlight=test-mode#adding-support-for-test-mode) to the track. Does it make sense to adjust the challenge so it'll run on a 1g heap ES?

File: so/challenges/default.json [hunk 2]
```diff
@@ -50,4 +50,111 @@
...
+        {
+          "name": "start-transform-group-by-hour",
+          "operation": {
+            "operation-type": "start-transform",
+            "transform-id": "transform-group-by-hour"
+          }
+        },
+        {
+          "name": "wait-for-transform-group-by-hour",
+          "operation": {
```
Comment (DJRickyB) [inline so/challenges/default.json:152]: If the point is to measure duration here, you will want 
```suggestion
          "include-in-reporting": true,
          "operation": {
```

File: so/challenges/default.json [hunk 3]
```diff
@@ -50,4 +50,111 @@
...
+        {
+          "name": "start-transform-group-by-user",
+          "operation": {
+            "operation-type": "start-transform",
+            "transform-id": "transform-group-by-user"
+          }
+        },
+        {
+          "name": "wait-for-transform-group-by-user",
+          "operation": {
```
Comment (DJRickyB) [inline so/challenges/default.json:125]: If the point is to measure duration here, you will want 
```suggestion
          "include-in-reporting": true,
          "operation": {
```

**Comment by DJRickyB** [comment]
You will need to merge latest master to your branch in order to pass CI

**Comment by inqueue** [comment]
> This PR does not add a track, but a challenge. I see a `1k` file as part of the track [files.txt](https://github.com/elastic/rally-tracks/blob/master/so/files.txt). Isn't that all we need?

Oops, I looked right past the `1k` file. Yes, that is all we need.

**Review by inqueue — APPROVED** [review]
> @DJRickyB could you run the full track to see if it stalls on heap consumption?

Rick mentioned this would be covered by the autogenerated tests in CI. This PR LGTM pending Rick's most recent commit suggestion to add reporting. Thanks!

**Comment by hendrikmuhs** [comment]
@elasticmachine please update branch

**Review by DJRickyB — APPROVED** [review]
LGTM

**Pull request merged**